gateway-js/src/executeQueryPlan.ts

```diff
@@ -686,9 +691,38 @@ function flattenResultsAtPath(value: any, path: ResponsePath): any {
   }
 }
 
+/**
+ * FIXME: this makes a fragile assumption that downstream paths in nested nodes
```

Comment:
My suggestion would be to pass the `FetchNode` down to this method so it has as much context it can. The `downstreamPath[0] === '_entities'` test could instead check if there is some `fetch.requires` (which is essentially the one way to know if a fetch is an entity fetch or not at the moment; we can probably improve that separately) and check if `downstreamPath[0] === '_entities'` in an assertion. This should make it more obvious if some future change the assumption that "fetch with requires/inputs == _entities call".